client/verta/tests/test_endpoint/test_endpoint.py

```diff
@@ -118,6 +121,35 @@ def test_repr(self, client, created_endpoints, experiment_run, model_for_deploym
         str_repr = repr(endpoint)
         assert "curl: {}".format(endpoint.get_deployed_model().get_curl()) in str_repr
 
+    def test_download_manifest(self, client, in_tempdir):
+        download_to_path = "manifest.yaml"
+        path = verta._internal_utils._utils.generate_default_name()
+        name = verta._internal_utils._utils.generate_default_name()
+
+        strategy = CanaryUpdateStrategy(interval=10, step=0.1)
+        strategy.add_rule(MaximumAverageLatencyThresholdRule(0.1))
+        resources = [CpuMillis(100), Memory("128Mi")]
+        autoscaling = Autoscaling(min_replicas=1, max_replicas=10, min_scale=0.1, max_scale=2)
+        autoscaling.add_metric(CpuUtilizationTarget(0.75))
+        env_vars = {'foo': "bar"}
+
+        filepath = client.download_endpoint_manifest(
+            download_to_path=download_to_path,
+            path=path,
+            name=name,
+            strategy=strategy,
+            autoscaling=autoscaling,
+            resources=resources,
+            env_vars=env_vars,
+        )
+
+        # can be loaded as YAML
+        with open(filepath, 'rb') as f:
+            manifest = yaml.safe_load(f)
+
+        assert manifest['kind'] == "Endpoint"
+        assert manifest['spec']['endpoint']['spec']['ingress']['path'] == path
```

Comment:
This will fail very soon when we add workspace to be part of the path =) I'd recommend checking something that is directly used by k8s, like env vars.